service/history/historyEngine.go

```diff
@@ -178,9 +179,18 @@ func NewEngineWithShardContext(
 		metricsClient:        shard.GetMetricsClient(),
 		historyEventNotifier: historyEventNotifier,
 		config:               config,
-		archivalClient:       archiver.NewClient(shard.GetMetricsClient(), shard.GetLogger(), publicClient, shard.GetConfig().NumArchiveSystemWorkflows, shard.GetConfig().ArchiveRequestRPS),
+		archivalClient:       warchiver.NewClient(shard.GetMetricsClient(), shard.GetLogger(), publicClient, shard.GetConfig().NumArchiveSystemWorkflows, shard.GetConfig().ArchiveRequestRPS),
 		archiverProvider:     archiverProvider,
 	}
+	historyArchiverBootstrapContainer := &archiver.HistoryBootstrapContainer{
+		HistoryManager:   historyManager,
+		HistoryV2Manager: historyV2Manager,
+		Logger:           shard.GetLogger(),
+		MetricsClient:    shard.GetMetricsClient(),
+		ClusterMetadata:  shard.GetClusterMetadata(),
+		DomainCache:      shard.GetDomainCache(),
+	}
+	historyEngImpl.archiverProvider.RegisterBootstrapContainer(common.HistoryServiceName, historyArchiverBootstrapContainer, &archiver.VisibilityBootstrapContainer{})
```

Comment:
I am not 100% sure whether the registration should live here or when a new history handler is created.

Reply:
It should live at the very top level - service/history/service.go::Start()